src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.NavigateToItemDisplay.cs

```diff
@@ -58,6 +58,11 @@ public ReadOnlyCollection<DescriptionItem> DescriptionItems
             private ReadOnlyCollection<DescriptionItem> CreateDescriptionItems()
             {
                 var document = _searchResult.NavigableItem.Document;
+                if (document == null)
```

Comment:
Yes, when the API becomes public we would need to check validity for everything within the result returned by external provider. Currently, we have an internal prototype built on top of this result provider API that lazily computes the Document, while requiring to synchronously return the navigable items - we may invariably hit error conditions while lazily computing the document and end up returning a null document. We do ensure that we return valid span though.

In short - the current checks suffice the internal requirements for these APIs but we **should** eventually harden against any value returned by a third party provider when we make the APIs public.